### PR TITLE
Doc: scopes importations with csvs

### PR DESCRIPTION
**Pull request by laurajaime**
#### :tophat: What? Why?

Add doc about how scopes importation works.

#### :pushpin: Related Issues
- Related to #?
- Fixes #?

#### :clipboard: Subtasks
- [ ] Add documentation regarding the feature 
- [ ] Add/modify seeds
- [ ] Add tests
- [ ] Another subtask

### :camera: Screenshots (optional)
Description

### CODE REVIEW

File: docs/scopes_importation.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+### **scopes_hierarchy.csv**
+
+```
+FR,FC,SL
+
+0002,5100,5117
+```
+In this example, we have relation between Girona's code and their local and comarcal codes. In this case, 5100 is the relate comarcal code and 5117 is the relate local code.
+
+Caption:
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
Legend:
```

File: docs/scopes_importation.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+**FR**: regional code
+**FC**: comarcal code
+**SL**: local code
+
+## Rake tasks for run
+
+Next step after have the files is execute rake tasks in order to generate required yml files.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
Next step after having the files is executing the following rake tasks in order to generate required yml files.
```

File: docs/scopes_importation.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+'0201': L'Alcora
+'0301': Alcoi
+'0402': Alcover
+```
+
+## 3. rake erc_auth:csv_import:regionals 
+
+This task generate the file `regionals.yml` from `scopes_codes.csv`.
+
+Check it the rows it has FR code and add to yml.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]: Can you please better explain? :smile:

File: docs/scopes_importation.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+'0002': Girona
+'0003': Camp de Tarragona
+'0004': Ebre
+```
+
+## 4. rake erc_auth:csv_import:local_comarcal_relationships
+
+This task generate the file `local_comarcal_relationships.yml` from the files `locals.yml`, `comarcals.yml` and `scopes_hierarchy.csv`.
+
+Search local code and in the same row, get FC (comarcal) code.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]: So the task searches the local code (from which file?) in the same row (same as what?), then gets the comarcal code, and....
Can you please better explain? :smile:

File: docs/scopes_importation.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+'0110': '0100'
+'0201': '0200'
+'0301': '0300'
+```
+
+## 5. rake erc_auth:csv_import:local_regional_relationships 
+
+This rake generate the file `local_regional_relationships.yml` from files `locals.yml`, `regionals.yml` and `scopes_hierarchy.csv`.
+
+Search local code and in the same row, get FR (regional) code.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]: Can you please better explain? :pray:

File: docs/scopes_importation.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+---
+'2100': Baix Llobregat
+'5600': Baix Besòs - Maresme
+```
+
+## 7. rake civi_crm:generate:decidim_scopes_mapping
+
+This task generate the file `decidim_scopes_mapping.yml` from files `comarcal_exceptions.yml`, `local_comarcal_relationships.yml` and `local_regional_relationships.yml`.
+
+This file is used for register users. For that, the code on the left in file is the local scope code in the csv users. The code on the right is a code field in a scope in Decidim.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
This file is used on user registration. For that, the code on the left in file is the local scope code in the csv users. The code on the right is a code field in a scope in Decidim.
```

File: docs/scopes_importation.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+Use case:
+
+An user has 0103 code in csv users -> this code is relate with 0010 in this file -> this code is relate with a code field of scope in Decidim -> user registered successfully!
```
Comment (tramuntanal) [inline docs/scopes_importation.md]: Maybe this explanation is clearer?
```suggestion
An user has code 0103 in csv users -> this code is associated with 0010 in file `decidim_scopes_mapping.yml` -> so 0010 should belong to the `code` field of scope in Decidim -> user registered successfully!
```
did I understood correctly?

**Review by tramuntanal — COMMENTED** [review]
@LauraJaimeC this is a great documentation that you created with Ariadna.

I have some questions about what some rake tasks do. Can you please further explain, thanks! :smiley:

File: docs/scopes_importation.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+rake civi_crm:generate:decidim_scopes_mapping
+rake civi_crm:create:scopes 
+```
+
+## 1. rake erc_auth:csv_import:comarcals
+
+This tasks generate the file `comarcals.yml` from `scopes_codes.csv`.
+
+Check the rows it has FC code and add to yml.
+
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
This tasks generates the file `comarcals.yml` from `scopes_codes.csv`.

Checks the rows in `scopes_codes.csv` that have FC code and adds them to `comarcals.yml`.
```

File: docs/scopes_importation.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+'0400': Alt Camp
+'0500': Alt Empordà
+'0600': Alt Maestrat
+```
+
+## 2. rake erc_auth:csv_import:locals
+
+This task generate the file `locals.yml` from `scopes_codes.csv`.
+
+Check the rows it has SL code and add to yml.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
This task generates the file `locals.yml` from `scopes_codes.csv`.

Checks the rows in `scopes_codes.csv` that have SL code and adds them to `locals.yml`.
```

File: docs/scopes_importation.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+'0201': L'Alcora
+'0301': Alcoi
+'0402': Alcover
+```
+
+## 3. rake erc_auth:csv_import:regionals 
+
+This task generate the file `regionals.yml` from `scopes_codes.csv`.
+
+Check it the rows it has FR code and add to yml.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
This tasks generates the file `regionals.yml` from `scopes_codes.csv`.

Checks the rows in `scopes_codes.csv` that have FR code and adds them to `regionals.yml`.
```

File: docs/scopes_importation.md [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+'0002': Girona
+'0003': Camp de Tarragona
+'0004': Ebre
+```
+
+## 4. rake erc_auth:csv_import:local_comarcal_relationships
+
+This task generate the file `local_comarcal_relationships.yml` from the files `locals.yml`, `comarcals.yml` and `scopes_hierarchy.csv`.
+
+Search local code and in the same row, get FC (comarcal) code.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
This task generates the file `local_comarcal_relationships.yml` from the files `locals.yml`, `comarcals.yml` and `scopes_hierarchy.csv`.

`comarcals.yml` is loaded but it seems that it is not being used.
The task iterates `locals.yml` and, for each row, it searches for the SL code in `scopes_hierarchy.csv` and takes the  FC (comarcal) code. Once the correspondence is resolved it is saved in the output `local_comarcal_relationships.yml` file.
```

File: docs/scopes_importation.md [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+'0110': '0100'
+'0201': '0200'
+'0301': '0300'
+```
+
+## 5. rake erc_auth:csv_import:local_regional_relationships 
+
+This rake generate the file `local_regional_relationships.yml` from files `locals.yml`, `regionals.yml` and `scopes_hierarchy.csv`.
+
+Search local code and in the same row, get FR (regional) code.
```
Comment (tramuntanal) [inline docs/scopes_importation.md]:
```suggestion
This rake generates the file `local_regional_relationships.yml` from files `locals.yml`, `regionals.yml` and `scopes_hierarchy.csv`.

`regionals.yml` is loaded but it seems that it is not being used.
The task iterates `locals.yml` and, for each row, it searches for the SL code in `scopes_hierarchy.csv` and takes the  Fr (regional) code. Once the correspondence is resolved it is saved in the output `local_regional_relationships.yml` file.
Search local code and in the same row, get FR (regional) code.
```

**Comment by laurajaime** [comment]
@tramuntanal changes applied!

**Pull request merged**